db/rdbms/src/test/java/io/camunda/db/rdbms/write/queue/DefaultExecutionQueueTest.java

```diff
@@ -200,4 +269,57 @@ public QueueItem merge(final QueueItem originalItem) {
     assertThat(executionQueue.getQueue().get(0)).isSameAs(item1);
     assertThat(executionQueue.getQueue().get(1)).isSameAs(item2);
   }
+
+  @Test
+  public void shouldSortQueueItemsDuringFlush() {
+    executionQueue.executeInQueue(
+        new QueueItem(
+            ContextType.PROCESS_INSTANCE,
+            WriteStatementType.UPDATE,
+            1L,
+            "statement1",
+            "parameter1"));
+    executionQueue.executeInQueue(
+        new QueueItem(
+            ContextType.PROCESS_INSTANCE,
+            WriteStatementType.INSERT,
+            1L,
+            "statement2",
+            "parameter2"));
+    executionQueue.executeInQueue(
+        new QueueItem(
+            ContextType.PROCESS_INSTANCE,
+            WriteStatementType.DELETE,
+            1L,
+            "statement3",
+            "parameter3"));
+    executionQueue.executeInQueue(
+        new QueueItem(
+            ContextType.FLOW_NODE, WriteStatementType.UPDATE, 1L, "statement4", "parameter4"));
+    executionQueue.executeInQueue(
+        new QueueItem(
+            ContextType.FLOW_NODE, WriteStatementType.INSERT, 1L, "statement5", "parameter5"));
+    executionQueue.executeInQueue(
+        new QueueItem(
+            ContextType.USER_TASK, WriteStatementType.DELETE, 1L, "statement6", "parameter6"));
+    executionQueue.executeInQueue(
+        new QueueItem(
+            ContextType.USER_TASK, WriteStatementType.INSERT, 1L, "statement7", "parameter7"));
+    executionQueue.executeInQueue(
+        new QueueItem(
+            ContextType.USER_TASK, WriteStatementType.DELETE, 1L, "statement8", "parameter8"));
+
+    // when
+    executionQueue.flush();
+
+    // then
+    verify(session).update(eq("statement5"), any());
+    verify(session).update(eq("statement2"), any());
+    verify(session).update(eq("statement6"), any());
+    verify(session).update(eq("statement7"), any());
+    verify(session).update(eq("statement8"), any());
+    verify(session).update(eq("statement4"), any());
+    verify(session).update(eq("statement1"), any());
+    verify(session).update(eq("statement3"), any());
```

Comment:
🔧 This doesn't actually check the order. I'd change it to an arg captor and verify the actual order.